pkg/tfvars/aws/aws.go

```diff
@@ -1,22 +1,63 @@
+// Package aws contains AWS-specific Terraform-variable logic.
 package aws
 
-// AWS converts AWS related config.
-type AWS struct {
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1alpha1"
+)
+
+type config struct {
 	EC2AMIOverride string            `json:"aws_ec2_ami_override,omitempty"`
 	ExtraTags      map[string]string `json:"aws_extra_tags,omitempty"`
-	Master         `json:",inline"`
-	Region         string `json:"aws_region,omitempty"`
+	EC2Type        string            `json:"aws_master_ec2_type,omitempty"`
+	IOPS           int64             `json:"aws_master_root_volume_iops,omitempty"`
+	Size           int64             `json:"aws_master_root_volume_size,omitempty"`
+	Type           string            `json:"aws_master_root_volume_type,omitempty"`
+	Region         string            `json:"aws_region,omitempty"`
 }
 
-// Master converts master related config.
-type Master struct {
-	EC2Type          string `json:"aws_master_ec2_type,omitempty"`
-	MasterRootVolume `json:",inline"`
-}
+// TFVars generates AWS-specific Terraform variables launching the cluster.
+func TFVars(masterConfig *v1alpha1.AWSMachineProviderConfig) ([]byte, error) {
+	tags := make(map[string]string, len(masterConfig.Tags))
+	for _, tag := range masterConfig.Tags {
+		tags[tag.Name] = tag.Value
+	}
+
+	cfg := &config{
+		Region:         masterConfig.Placement.Region,
+		ExtraTags:      tags,
+		EC2AMIOverride: *masterConfig.AMI.ID,
+		EC2Type:        masterConfig.InstanceType,
+	}
+
+	if len(masterConfig.BlockDevices) == 0 {
+		return nil, errors.New("block device slice cannot be empty")
+	}
+
+	rootVolume := masterConfig.BlockDevices[0]
+	if rootVolume.EBS == nil {
+		return nil, errors.New("EBS information must be configured for the root volume")
+	}
+
+	if rootVolume.EBS.Iops == nil {
+		cfg.IOPS = 400
```

Comment:
i don't think these defaults must exist in this package...

Reply:
> i don't think these defaults must exist in this package...

Did you want me to just assume they were set and panic if they aren't?

Reply:
expect them to be set and error otherwise, maybe ?

Reply:
> expect them to be set and error otherwise, maybe ?

Done with 42a8a10 -> 6d449c5.